server/entrypoint.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -o nounset
+set -o errexit
+
+UID=$(id -u)
+if ! whoami &> /dev/null; then
+  if [ -w /etc/passwd ]; then
+    echo "Creating mesh user uid to $UID"
+    echo "${USER_NAME:-mesh}:x:$UID:0:${USER_NAME:-mesh} user:/mesh:/sbin/nologin" >> /etc/passwd
```

Comment:
...which is then used here to create a new user with the same ID. Won't this result in two users with the same ID?

Reply:
The user with the given Id may not have a passwd entry. Thus whoami would fail. If whoami fails we need to create the entry in passwd to assign a "username" to this UID. Otherwise the process would just run with a user without a "real" name.